examples/Locomotion/Teleportation_Colliders/main.lua

```diff
@@ -0,0 +1,166 @@
+  local handPose = mat4(motion.pose):mul(mat4(lovr.headset.getPose('right')))
+  local handPosition = vec3(handPose)
+  local handDirection = quat(handPose):direction()
+  -- Intersect with world geometry by casting a ray
+  local origin = vec3(handPose:mul(0, 0, -0.2))
+  local target = vec3(handPose:mul(0, 0, -motion.teleportDistance))
+  local intersectionDistance = math.huge
+  local closestHit
+  physicsWorld:raycast(origin, target,
+    function(shape, x, y, z, nx, ny, nz)
+      local position = vec3(x, y, z)
+      local normal = vec3(nx, ny, nz)
+      local distance = (origin - position):length()
+      -- Teleportation is possible if distance is within range and if surface is roughly horizontal
+      if distance < intersectionDistance and normal:dot(vec3(0,1,0)) > 0.5 then
+        intersectionDistance = distance
+        closestHit = position
+      end
+    end)
+  if closestHit then
+    motion.teleportValid = true
+    motion.targetPosition:set(closestHit)
+  else
+    motion.teleportValid = false
+    motion.targetPosition:set(handPose:mul(0,0,-20))
+  end
+  -- Construct teleporter visualization curve 
+  local midPoint = vec3(handPosition):lerp(motion.targetPosition, 0.3)
+  if motion.teleportValid then
+    midPoint:add(vec3(0, 0.1 * intersectionDistance, 0)) -- Fake a parabola
+  end
+  motion.teleportCurve:setPoint(1, handPosition)
+  motion.teleportCurve:setPoint(2, midPoint)
+  motion.teleportCurve:setPoint(3, motion.targetPosition)
+  -- Start timer when jump is triggered, preform jump on half-blink
+  if lovr.headset.wasPressed('right', 'trigger') and motion.teleportValid then
+    motion.blinkStopwatch = -motion.blinkTime / 2
+  end
+  -- Preform jump with VR pose offset by relative distance between here and there
+  if motion.blinkStopwatch < 0 and
+     motion.blinkStopwatch + dt >= 0 then
+    local headsetXZ = vec3(lovr.headset.getPosition())
+    headsetXZ.y = 0
+    local newPosition = motion.targetPosition - headsetXZ
+    motion.pose:set(newPosition, vec3(1,1,1), quat(motion.pose)) -- ZAAPP
+  end
+  -- Snap horizontal turning (often combined with teleport mechanics)
+  if lovr.headset.isTracked('right') then
+    local x, y = lovr.headset.getAxis('right', 'thumbstick')
+    if math.abs(x) > motion.thumbstickDeadzone and motion.thumbstickCooldown < 0 then
+      local angle = -x / math.abs(x) * motion.snapTurnAngle
+      motion.pose:rotate(angle, 0, 1, 0)
+      motion.thumbstickCooldown = motion.thumbstickCooldownTime
+    end
+  end
+  motion.blinkStopwatch = motion.blinkStopwatch + dt
+  motion.thumbstickCooldown = motion.thumbstickCooldown - dt
+end
+
+function motion.drawTeleport()
+  if lovr.headset.isTracked('right') then
+    -- Teleport target and curve
+    lovr.graphics.setColor(1, 1, 1, 0.1)
+    if motion.teleportValid then
+      lovr.graphics.setColor(1, 1, 0)
+      lovr.graphics.cylinder(motion.targetPosition, 0.05, math.pi/2,  1,0,0,  0.4, 0.4)
```

Comment:
I made this change during testing so this cylinder doesn't clip into the colliders

```lua
local height = 0.05
local targetPosition = motion.targetPosition + vec3(0, height / 2, 0)
lovr.graphics.cylinder(targetPosition, height, math.pi/2,  1,0,0,  0.4, 0.4)
```

Very minor, feel free to discard